### ISSUE TITLE
Possible unintended change in convergence behavior

### ISSUE THREAD

**Issue by michaelmckinsey1** [LBANN/ScaFFold#96]
We overlooked a change when reviewing #88 to convergence behvaior. The validation dice calculation changed to argmax instead of softmax (F05 from the `REVIEW.md` document).

Argmax removes the constraint for the model to improve prediction confidence beyond majority. For example, how the validation dice is measured now vs originally for two cases (both background majority prediction):
```
CURRENT (#88/#92, hard val dice)
  background: 0.51, foreground: 0.49
  train loss: 1.673, val loss: 1.673, train dice: 0.000002, val dice: 1.000000

  CURRENT (#88/#92, hard val dice)
  background: 0.99, foreground: 0.01
  train loss: 1.010, val loss: 1.010, train dice: 0.000100, val dice: 1.000000

  ORIGINAL (soft val dice)
  background: 0.51, foreground: 0.49
  train loss: 1.673, val loss: 1.673, train dice: 0.000002, val dice: 0.000002

  ORIGINAL (soft val dice)
  background: 0.99, foreground: 0.01
  train loss: 1.010, val loss: 1.010, train dice: 0.000100, val dice: 0.000100
```
(train loss, val loss, and train dice all softmax still).

We must decide which metric we want to use, as using argmax or softmax have different meanings for this benchmark. Argmax will obviously converge faster than softmax.

<img width="729" height="891" alt="Image" src="https://github.com/user-attachments/assets/f97ae058-c339-4717-ad5e-0f8e99627140" />

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/722e226f-7b26-4e7c-9eef-f4bd7fb7bef1" />

**Comment by michaelmckinsey1** [comment]
We have decided that, although the resulting model is clearly different using the argmax convergence criteria, since we are benchmarking faster convergence is a desired property in this case. Furthermore, the argmax criteria does not affect our ability to benchmark the software stacks and kernels. (@ndryden)
